File: front/src/Entities/Room.ts
import { Moment } from "moment"
import User from "./User"
import Message from "./Message"

class Room {
    id: number
    name: string
    creator: User
    dateCreated: Moment

    constructor( 
        id: number,
        name: string,
        creator: User,
        dateCreated: Moment
    ){
        this.id = id
        this.name = name
        this.creator = creator
        this.dateCreated = dateCreated
    }
}

export default Room

File: Back/src/main/java/com/playful/chat/model/Room.java
package com.playful.chat.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "creator_id")
    private UserModel creator;

    private int totalUsers;

    private LocalDateTime dateCreated;

}